stdlib/public/core/RangeReplaceableCollectionType.swift

```diff
@@ -459,7 +459,7 @@ public func +<
     where S.Generator.Element == C.Generator.Element
 >(lhs: S, rhs: C) -> C {
   var result = C()
-  result.reserveCapacity(rhs.count + numericCast(rhs.underestimateCount()))
+  result.reserveCapacity(rhs.count + numericCast(lhs.underestimateCount()))
```

Comment:
This fix LGTM.  Feel free to submit it as a separate PR and I'll merge immediately.